### PR TITLE
aws-crt-cpp: add missing interface definition if shared

### PR DESCRIPTION
**Pull request by SpaceIm**
see https://github.com/awslabs/aws-crt-cpp/blob/v0.19.9/include/aws/crt/Exports.h

closes https://github.com/conan-io/conan-center-index/issues/17382

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
waiting for https://github.com/conan-io/conan-center-index/pull/17100, https://github.com/conan-io/conan-center-index/pull/17104, https://github.com/conan-io/conan-center-index/pull/17105, https://github.com/conan-io/conan-center-index/pull/17107, https://github.com/conan-io/conan-center-index/pull/17108, https://github.com/conan-io/conan-center-index/pull/17110, https://github.com/conan-io/conan-center-index/pull/17111, https://github.com/conan-io/conan-center-index/pull/17117 & https://github.com/conan-io/conan-center-index/pull/17119

**Review by jwillikers — APPROVED** [review]

File: recipes/aws-crt-cpp/all/conanfile.py [hunk 1]
```diff
@@ -46,22 +47,23 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("aws-c-common/0.8.2")
+        self.requires("aws-c-cal/0.5.13", transitive_headers=True)
```
Comment (mpimenov) [inline recipes/aws-crt-cpp/all/conanfile.py:50]: Out of curiosity — can these transitive_headers additions be dropped after all libraries from https://github.com/conan-io/conan-center-index/pull/17125 are aligned?

**Review by mpimenov — APPROVED** [review]

File: recipes/aws-crt-cpp/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/aws-crt-cpp/all/conanfile.py:50]: No, these headers are really transitive.

**Comment by mpimenov** [comment]
@jcar87 Could you look at this please?

**Comment by jcar87** [comment]
Relaunched CI jobs for this, will check later to complete the review.

**Comment by jcar87** [comment]
Waiting for https://github.com/conan-io/conan-center-index/pull/17117 to be merged before starting this one.

**Comment by jcar87** [comment]
Relaunched job after promotion of missing packages

**Comment by SpaceIm** [comment]
Thanks

**Comment by mpimenov** [comment]
Any progress on the review? This is the last one blocking #17125

**Comment by SpaceIm** [comment]
@jcar87 @uilianries @prince-chrismc @RubenRBS could you review please?

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**